mechanics/model/Battle.py

```diff
@@ -46,6 +58,13 @@ def _get_entry_from_queue(self) -> PrioritizedItem:
 
         return entry
 
+    def _get_nearest_defending_token(self, attacking_token) -> Token:
+        while True:
+            distance, token = self.distance_to_opponents[attacking_token.unit.id][0]
+            if token.unit.alive:
+                return token
+            self.distance_to_opponents[attacking_token.unit.id].pop(0)
```

Comment:
to me, that is not "pythonic" code, because it's very stateful and not functional, but let's keep it as it is for now